pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -390,6 +397,43 @@ void writeMetadata()
     properties.save();
   }
 
+  /**
+   * Check for the validity of segment start and end time
+   * @param startTime segment start time
+   * @param endTime segment end time
+   * @param segmentName segment name
+   */
+  private void checkTime(final Object startTime, final Object endTime, final String segmentName) {
+    if (startTime == null || endTime == null) {
+      throw new RuntimeException("Expecting non-null start/end time for segment: " + segmentName);
+    }
+
+    long startMillis = 0;
+    long endMillis = 0;
+
+    if (startTime instanceof Long && endTime instanceof Long) {
+      startMillis = (long)startTime;
+      endMillis = (long)endTime;
+    } else if (startTime instanceof Integer && endTime instanceof Integer) {
+      startMillis = ((Integer) startTime).longValue();
+      endMillis = ((Integer) endTime).longValue();
+    } else if (startTime instanceof String && endTime instanceof String) {
+      startMillis = Long.parseLong((String)startTime);
+      endMillis = Long.parseLong((String)endTime);
```

Comment:
I am not quite sure about the type comparison. Have some questions on the format of time that is incoming here at the time of generation v/s how it is stored. By looking at the code in SegmentMetadataImpl, seems like it is stored as string and is always parsed as long